src/lib/db.ts

```diff
@@ -89,6 +111,7 @@ export class DB {
   // adds data at `shared-tables/${shareId}/${key}`
   set(key: string, data: any) {
     if (this.shareRef) {
+      removeIds(data);
```

Comment:
My concern here is that there are some IDs we don't want to remove -- notably the persistent IDs of items, attributes, and now data contexts. For attributes and data contexts the IDs we want to preserve are in fields with different names (but that isn't immediately obvious) and it seems that this function isn't currently called for items, but there's nothing to suggest that it couldn't/shouldn't be. I would suggest a comment indicating appropriate usage of the function and down the road could imagine wanting an optional argument to disable the removal at the client's discretion.

Reply:
Unsure what the takeaway is. Do you foresee cases where removing ids and guids will cause a problem?

Reply:
I think a comment could make it clear that the goal is to remove CODAP/SproutCore-local IDs, but that we don't want to remove the universal IDs that are to be used for synchronization, and so this function will not do the right thing if used to store items, for instance.

Reply:
Expanded on the existing comment and renamed the function. I feel like you have a better sense of the context as you're dealing with the uuids in items, so feel free to modify this further in your own branch if needed.